src/microbe_stage/PatchRegion.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   A region is a something like a continent/ocean that contains multiple biomes(patches).
+/// </summary>
+[UseThriveSerializer]
+public class PatchRegion
+{
+    [JsonProperty]
+    public readonly int ID;
+
+    [JsonIgnore]
+    public readonly ISet<PatchRegion> Adjacent = new HashSet<PatchRegion>();
+
+    [JsonIgnore]
+    public float PatchNodeWidth = 64.0f;
+
+    [JsonIgnore]
+    public float PatchNodeHeight = 64.0f;
+
+    [JsonIgnore]
+    public float RegionLineWidth = 4f;
+
+    [JsonIgnore]
+    public float PatchMargin = 4f;
+
+    [JsonProperty]
+    public string RegionType;
+
+    [JsonProperty]
+    public float Height;
+
+    [JsonProperty]
+    public float Width;
+
+    [JsonProperty]
+    public List<Patch> Patches;
```

Comment:
I think these should be all properties (and the readonlies should have private setters).